silences-gitops/README.md

```diff
@@ -0,0 +1,114 @@
+
+This solution is laid out as follows:
+
+- Silence CRs common to all clusters are stored in the MCB repository, under the `bases/silences/` directory
+- Silence CRs added in MCB also need to be referenced in the `bases/silences/kustomization.yaml` file
+- Silence CRs specific to a cluster are stored in the corresponding CMC repository, under the `management-clusters/<management-cluster-name>/silences/` directory
+- Name collision between MCB and CMC Silences CRs is prevented by adding the special `global-` name prefix when deploying Silences CRs from the MCB repository
+
+![Silences GitOps Architecture](silences-gitops.png)
+
+### Examples
+
+- Common Silence CR in MCB repository:
+
+```yaml
+$ cat management-cluster-bases/bases/silences/alice.yaml
+apiVersion: monitoring.giantswarm.io/v1alpha1
+kind: Silence
+metadata:
+  name: alice
+  annotations:
+    valid-until: "2100-01-01"
+spec:
+  matchers:
+  - name: alertname
+    value: AliceAlert
+    isRegex: false
+```
+
+```yaml
+$ cat management-cluster-bases/bases/silences/kustomization.yaml
+resources:
+- alice.yaml
+```
+
+- Cluster specific Silence CR in CMC repository:
+
+```yaml
+$ cat giantswarm-management-clusters/management-clusters/some-management-cluster/silences/bob.yaml
+apiVersion: monitoring.giantswarm.io/v1alpha1
+kind: Silence
+metadata:
+  name: bob
+  annotations:
+    valid-until: "2100-01-01"
+spec:
+  matchers:
+  - name: alertname
+    value: BobAlert
+    isRegex: false
+  - name: cluster_id
+    value: some-cluster-id
+    isRegex: false
+```
+
+### Use cases
+
+- Create a silence for all clusters
+  - Add the Silence CR to MCB repository in `bases/silences/` directory
+  - Reference the new Silence CR in the `bases/silences/kustomization.yaml` file
+
+- Create a silence for a single cluster
+  - Add the Silence CR to the `management-clusters/some-management-cluster/silences/` directory
+  - Add a matcher for the `cluster_id` tag with the cluster ID as value in the Silence CR
+
+- Create a silence for multiple clusters in a single management cluster
+  - Add the Silence CR to the `management-clusters/some-management-cluster/silences/` directory
+  - (Optionally) Add a matcher for the `cluster_id` tag with all cluster ID as regex value in the Silence CR
+
+- Create a silence for multiple clusters in a multiple management clusters
+  - Add the Silence CR for each desired management cluster in `management-clusters/<each-management-cluster>/silences/` directory
+  - (Optionally) Add a matcher for the `cluster_id` tag in the Silence CR
+
+### Benefits
+
+- Silences are deployed and managed using standard GitOps mechanisms
+- Customers can manage their own silences
+- Fixes current privacy and security issues
```

Comment:
Is there anything we cn get rid of?

Reply:
Like the cronjob?

Reply:
I added this in the `What changes` section, IMO that's not a benefit for the end user.

Reply:
Oh I missed that